### ISSUE TITLE
Check MPI 4 Info behavior: ignored keys should not be returned

### ISSUE THREAD

**Issue by devreal** [open-mpi/ompi#9555]
MPI 4 mandates that only info keys  not ignored by the implementation are returned by `MPI_[Win|File|Comm]_get_info`, e.g., 

> An MPI implementation is required to return all hints that are supported by the implementation and have default values specified; any user-supplied hints that were not ignored by the implementation; and any additional hints that were set by the implementation.

Following a discussion on the MPI Forum mailing list [1], we should make sure that OMPI drops all info keys it does not know during file/window/comm construction. Recent changes to the info subscriber infrastructure should ensure that this is already true for `MPI_[Win|File|Comm]_set_info` (https://github.com/open-mpi/ompi/pull/9474), but better double check. 

A reproducer from the mailing list discussion that should fail but does not in current master (and likely everywhere else):

```C
        #include <mpi.h>
        #include <stdlib.h>
        #include <stdio.h>
        #include <string.h>

        int main(void) {
            MPI_Init(NULL, NULL);

            MPI_Info info;
            MPI_Info_create(&info);
            MPI_Info_set(info, "foo", "bar");

            MPI_File fh;
            MPI_File_open(MPI_COMM_SELF, "testfile", MPI_MODE_RDWR | MPI_MODE_CREATE, info, &fh);

            MPI_Info info_used;
            MPI_File_get_info(fh, &info_used);

            char value[4] = "";
            int flag;
            MPI_Info_get(info_used, "foo", 3, value, &flag);

            if (!flag) {
                printf("key not found\n");
                MPI_Abort(MPI_COMM_WORLD, 1);
            }

            if (strcmp(value, "bar")) {
                printf("value does not match\n");
                MPI_Abort(MPI_COMM_WORLD, 1);
            }

            MPI_File_close(&fh);
            MPI_Info_free(&info);
            MPI_Info_free(&info_used);

            MPI_Finalize();
            return 0;
        }
```

[1] https://lists.mpi-forum.org/pipermail/mpi-forum/2021-October/007425.html

**Comment by hppritcha** [comment]
@devreal I think we need PR #9567  applied to v5.0.x branch as well.

**Comment by devreal** [comment]
@hppritcha This was merged into v5.0.x in https://github.com/open-mpi/ompi/pull/9770 (it didn't reference the issue though, sorry)
